### PR TITLE
jj: include default features

### PR DESCRIPTION
**Pull request by chriskrycho**
At the moment, this only includes specifically the `watchman` feature, which can dramatically improve performance on large repositories; and the project is pretty careful about adding new features to the list of defaults.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
Don't you need
```ruby
depends_on "watchman"
```
to use the `watchman` feature?

**Comment by chriskrycho** [comment]
Ah, good call! I'll update that. I have had watchman installed on my system for so long I half forgot it isn't built in. 😂

**Comment by chriskrycho** [comment]
Updated—thanks again for the feedback there, @carlocab!

**Review by ckoehler — COMMENTED** [review]
LGTM!

File: Formula/j/jj.rb [hunk 1]
```diff
@@ -20,12 +21,13 @@ class Jj < Formula
...
   depends_on "libgit2"
   depends_on "openssl@3"
+  depends_on "watchman"
   uses_from_macos "zlib"
 
   def install
     ENV["LIBGIT2_NO_VENDOR"] = "1"
 
-    system "cargo", "install", "--no-default-features", "--bin", "jj", *std_cargo_args(path: "cli")
+    system "cargo", "install", "--bin", "jj", *std_cargo_args(path: "cli")
```
Comment (chenrui333) [inline Formula/j/jj.rb:30]:
```suggestion
    system "cargo", "install", *std_cargo_args(path: "cli")
```

**Comment by chenrui333** [comment]
@chriskrycho can you also squash the commits and forced update your branch? Thanks!

**Comment by chriskrycho** [comment]
@chenrui333 done, and thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @chriskrycho!

**Comment by Aeron** [comment]
@chriskrycho, can this change be optional, like `with-watchman`? It was a small and self-contained binary two days ago, and now it brings the watchman and all its dependencies, which might not be a desirable behavior for some folks. It’s surprising, at least. And not everybody works with large repos.

Or vice-versa: if it’s crucial for a dominant part of the user base, maybe an option to have it raw and featureless is more suitable. Yes, more options are not always better for UX, but (kind of) breaking packaging (or delivery) changes aren’t as well.

I’m unsure if it deserves a separate issue, so I thought it’s better to address it here.

**Comment by carlocab** [comment]
We don't allow options in Homebrew/core (because they are infeasible to test properly in CI). If you want a version of this formula with options (e.g. `--with-watchman`), then that will have to be hosted in a third-party tap.

So the options here are:
1. Undo this change
2. Keep it as-is for all users
3. Depending on how `jj` behaves when it is built with the `watchman` feature but is unable to find a `watchman` server or binary, we could maybe keep the feature but drop the `watchman` dependency. But most Homebrew maintainers tend to frown upon shipping features in formulae without the required dependencies.

**Comment by Aeron** [comment]
Oh, I see. I overlooked the explicit note about it in the formula cookbook—that’s my bad.

I’m not sure what is the best course of action here. I’m not a maintainer or anything, just a “concerned citizen” who didn’t expect to get an extra 700 MB of dependencies.

Maybe it would be a viable solution for the Jujutsu repo to have a separate tap with options and leave the core variant in the most suitable form for most users. But I have no idea what form it is—maybe everyone wants the watchman.

**Comment by chriskrycho** [comment]
Thanks for flagging this up, @Aeron! It’s a fair point. (As I noted up-thread, I have had watchman installed on every machine I have had for a *very* long time now, so I didn’t even realize how large it is!) I’ll discuss it with the jj team and see what the best path forward is. Thanks @carlocab for the clear set of options!

**Comment by gpanders** [comment]
> Depending on how jj behaves when it is built with the watchman feature but is unable to find a watchman server or binary, we could maybe keep the feature but drop the watchman dependency. But most Homebrew maintainers tend to frown upon shipping features in formulae without the required dependencies.

Speaking also as a “concerned citizen” (to use @Aeron’s words), this seems like the most reasonable approach to me. jj is built with watchman support but does not (to my knowledge) actually depend on watchman to function properly unless the behavior is explicitly opted into. Just as git optionally can use watchman, but does not require it, and git does not have a dependency on watchman in Homebrew either.

**Comment by chriskrycho** [comment]
> jj is built with watchman support but does not (to my knowledge) actually depend on watchman to function properly unless the behavior is explicitly opted into.

That's correct! And I probably should have clarified as much in response to @carlocab’s original question about it that `jj` works fine with the feature enabled as part of its build but without `watchman` installed. I’ll open a PR to drop that part of it. Thanks again for bringing it up, @Aeron!

**Comment by chriskrycho** [comment]
I just opened #171735 in line with this discussion! 🎉

**Pull request merged**